code/framework/api-pub-sub-jetty/src/main/java/io/cattle/platform/api/pubsub/subscribe/jetty/JettyWebSocketSubcriptionHandler.java

```diff
@@ -37,20 +38,17 @@ protected MessageWriter getMessageWriter(ApiRequest apiRequest) throws IOExcepti
         HttpServletRequest req = apiRequest.getServletContext().getRequest();
         HttpServletResponse resp = apiRequest.getServletContext().getResponse();
         final WebSocketMessageWriter messageWriter = new WebSocketMessageWriter();
-
-        WebSocketFactory factory = new WebSocketFactory(new Acceptor() {
+        WebSocketServerFactory factory = new WebSocketServerFactory();
+        factory.getPolicy().setAsyncWriteTimeout(1000);
+        factory.setCreator(new WebSocketCreator() {
             @Override
-            public WebSocket doWebSocketConnect(HttpServletRequest request, String protocol) {
+            public Object createWebSocket(ServletUpgradeRequest req, ServletUpgradeResponse resp) {
                 return messageWriter;
             }
-
-            @Override
-            public boolean checkOrigin(HttpServletRequest request, String origin) {
-                return true;
-            }
         });
 
-        if (factory.acceptWebSocket(req, resp)) {
+        if ("websocket".equalsIgnoreCase(req.getHeader("Upgrade")) && factory.acceptWebSocket(req, resp)) {
```

Comment:
Effectively, it is already tested. The websocket specification requires that that header be present when a client requests a websocket connection. We already have tests that establish websocket connections.